### ISSUE TITLE
entry callbacks called 1 level shifted

### ISSUE THREAD

**Issue by EverydayDynamics** [thejpster/menu#20]
Hello, 
I've noticed that the entry (but not the exit) function calls don't line up with the actual sub-menu that's being entered.
This menu tree:
```Rust
let menu = Menu {
            label: "ROOT",
            items: &[
                &Item {
                    command: "sub1",
                    help: None,
                    item_type: ItemType::Menu(&Menu {
                        label: "",
                        items: &[
                            &Item {
                                command: "sub2",
                                help: None,
                                item_type: ItemType::Menu(&Menu {
                                    label: "",
                                    items: &[
                                    ],
                                    entry: Some(sub2_entry),
                                    exit: Some(sub2_exit),
                                }),
                            }
                        ],
                        entry: Some(sub1_entry),
                        exit: Some(sub1_exit),
                    }),
                }
            ],
            entry: Some(root_entry),
            exit: Some(root_exit),
        };
```
Produces the following output/ calls:
```
root entry
sub1
root entry
sub2
sub1 entry
exit
sub2 exit
exit
sub1 exit
```
To me the expected call order would be:
```
root entry
sub1
sub1 entry
sub2
sub2 entry
exit
sub2 exit
exit
sub1 exit
```
Here are the callback definitions for reference:
```Rust
fn root_entry<MIO: Reader+Write>(
    _menu: &Menu<MIO, Context>,
    interface: &mut MIO,
    context: &mut Context,
) {
    interface.write_str("root entry").unwrap()
}
fn root_exit<MIO: Reader+Write>(
    _menu: &Menu<MIO, Context>,
    interface: &mut MIO,
    context: &mut Context,
) {
    interface.write_str("root exit").unwrap()
}
fn sub1_entry<MIO: Reader+Write>(
    _menu: &Menu<MIO, Context>,
    interface: &mut MIO,
    context: &mut Context,
) {
    interface.write_str("sub1 entry").unwrap()
}
fn sub1_exit<MIO: Reader+Write>(
    _menu: &Menu<MIO, Context>,
    interface: &mut MIO,
    context: &mut Context,
) {
    interface.write_str("sub1 exit").unwrap()
}
fn sub2_entry<MIO: Reader+Write>(
    _menu: &Menu<MIO, Context>,
    interface: &mut MIO,
    context: &mut Context,
) {
    interface.write_str("sub2 entry").unwrap()
}
fn sub2_exit<MIO: Reader+Write>(
    _menu: &Menu<MIO, Context>,
    interface: &mut MIO,
    context: &mut Context,
) {
    interface.write_str("sub2 exit").unwrap()
}
```

**Comment by thejpster** [comment]
yeah, that sounds broken

**Comment by thejpster** [comment]
I think the bug is here: https://github.com/rust-embedded-community/menu/blob/master/src/lib.rs#L421

I call the entry function for the current menu before pushing the new one. I should push the new one first and then call the entry function.

**Comment by thejpster** [comment]
Feel free to submit a PR for this

**Comment by EverydayDynamics** [comment]
Cheers, I'll do it later today

**Comment by thejpster** [comment]
Fixed in #21
